apps/backend/src/libs/modules/token/base-token.module.ts

```diff
@@ -19,12 +19,19 @@ class BaseToken implements Token {
 		this.algorithm = algorithm;
 	}
 
-	public async createToken(payload: JWTPayload): Promise<string> {
-		return await new SignJWT(payload)
+	public async createToken(
+		payload: JWTPayload,
+		hasExpirationTime: boolean = true,
```

Comment:
Let's create a separate instance of BaseToken for projectApiKey, then we won't need to pass `hasExpirationTime`, we just won't pass the expiration time during the instance creation.
```suggestion
```